enhancements/accelerators/out-of-tree-drivers.md

```diff
@@ -0,0 +1,976 @@
+---
+title: Supporting out-of-tree drivers on OpenShift
+  - "@zvonkok"
+reviewers:
+  - "@ashcrow"
+  - "@darkmuggle"
+  - "@cgwalter"
+approvers:
+  - "@ashcrow"
+  - "@cgwalters"
+  - "@darkmuggle"
+creation-date: 2020-04-03
+last-updated: 2020-07-21
+status: provisional
+see-also:
+  - "/enhancements/TODO.md"
+replaces:
+  - "/enhancements/TODO.md"
+superseded-by:
+  - "/enhancements/TODO.md"
+---
+
+# Supporting out-of-tree drivers on OpenShift
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+1. What is the best way to find out which node belongs to which OpenShift
+release (imagine stuck nodes during ugrades)? See also email-thread on
+coreos-devel "Can we update os-release to reflect the "full" version
+of OpenShift?"
+2. Can an OLM managed operator create an ClusterOperator object to manage
+upgradeability of the cluster? SRO needs to run some preflight checks
+upfront before MCO updates the nodes to make sure the new kernel or
+driver-container is compatible.
```

Comment:
@smarterclayton  Can an OLM managed operator create a ClusterOperator object to manage
upgradeability of the cluster? In the case of SRO we do not want to prevent SRO updates, rather we want to make sure the new MCO(+kernel) is compatible with the out-of-tree driver.